usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
illumos also has a limited queue size for sysevents, and flooding sysevents can lead to lost to events, which can lead to other badness.   sysevents are a great thing, but we must have some care not to overwhelm any user space subscriber.  I think the problem here is snapshots.